lib/krb5/store.c

```diff
@@ -335,17 +336,62 @@ krb5_storage_to_data(krb5_storage *sp, krb5_data *data)
     return 0;
 }
 
+static size_t
+pack_int(void *buffer, uint64_t value)
+{
```

Comment:
Two issues here: firstly, 32-bit negative numbers (e.g. ~0UL) are sign extended which makes the encoding less efficient. We could add a size argument to `pack_int()` and mask `value` off, or perhaps there is another approach.

Secondly, encoding ~0ULL is not possible – the decoder parses it as two integers. We will need to expand the number of bits used to encode the length. If the length can never be zero, then we can encode one less than the length, but we still need to account for `length - 1 == 8` for ~0ULL.

Cc @nicowilliams